src/core/service.c

```diff
@@ -1482,36 +1482,29 @@ static int service_create_monitor_md_env(Job *j, char **ret) {
                 if (!env_source)
                         continue;
 
-                if (first) {
+                if (first)
                         /* Add the environment variable name first. */
-                        r = strextendf(&var, "MONITOR_METADATA=");
-                        if (r < 0)
-                                return r;
-
-                }
+                        if (!strextend(&var, "MONITOR_METADATA="))
```

Comment:
`first && !strextend`?

Reply:
generally I find it more legible to avoid mixing logic checks and error checks, but it's not too important, fixed